controllers/yawol-controller/loadbalancer/loadbalancer_controller.go

```diff
@@ -421,17 +422,20 @@ func (r *Reconciler) reconcileFIP(
 	// Get FIP
 	var fip *floatingips.FloatingIP
 	if fip, err = openstackhelper.GetFIPByID(ctx, fipClient, *lb.Status.FloatingID); err != nil {
-		switch err.(type) {
-		case gophercloud.ErrUnexpectedResponseCode, gophercloud.ErrResourceNotFound:
-			r.Log.Info("fip not found in openstack", "fip", *lb.Status.FloatingID)
-			// fip not found by ID, remove it from status and trigger reconcile
-			if err := helper.RemoveFromLBStatus(ctx, r.Status(), lb, "floatingID"); err != nil {
-				return false, fmt.Errorf("failed to remove FIP from lb status: %w", err)
-			}
-			return true, err
-		default:
-			r.Log.Info("unexpected error occurred")
-			return false, kubernetes.SendErrorAsEvent(r.RecorderLB, err, lb)
+		switch e := err.(type) {
+			case gophercloud.ErrUnexpectedResponseCode:
+				code := e.GetStatusCode()
+				if code == 404 {
+					// FIP not found by ID. Unable to determine if is's due to an OpenStack API outage or a legitimate issue.
+					// legitimate issue.
+					r.Log.Info("fip not found in openstack", "fip", *lb.Status.FloatingID)
+				} else {
+					r.Log.Info("unexpected gophercloud error occurred", "error: ", err)
```

Comment:
use `r.Log.Error` instead

Reply:
Do we need to log it there? Or is the error we return also logged at the end?

Reply:
will remove